monteCarloTreeSearch.c

```diff
@@ -0,0 +1,334 @@
+				if(numOfActions != 0)
+				{
+					//Randomly select a move to make
+					randomValue = numOfActions / 2;
+					randomValue = rand() % randomValue + 1;  //may not be plus 1
+					boardMove[0] = allActions[randomValue];
+					boardMove[1] = allActions[randomValue + 1]; 
+				}
+				else
+				{
+					boardMove[0] = -1;
+					boardMove[1] = -1;
+				}
+				//player cannot make a move
+				if(boardMove[0] == -1 && boardMove[1] == -1)
+				{
+
+					//Check if other player can move
+					flipMove(boardState->whoseMove);
+					pickMove(boardState, nodeLimit, timeLimit, boardMove);
+
+					//determine the winner because other player cannot make a move either
+					if(boardMove[0] == -1 && boardMove[1] == -1)
+					{
+						gameContinues = -10;
+						winner = determineWinner(boardState);
+						if(winner == 0)
+						{
+							selected->numSimulations += 1;
+
+						}
+						else if(winner == 1)
+						{
+							if(boardState->whoseMove == 1)
+							{
+								selected->numWins += 1;
+							}
+							selected->numSimulations += 1;
+
+						}	
+						else if(winner == 2)
+						{
+							if(boardState->whoseMove == 2)
+							{
+								selected->numWins += 1;
+							}
+							selected->numSimulations += 1;
+
+						}
+					}	
+					//Continue the game
+					else
+					{
+						placePiece(boardState, boardMove);
+					}
+				}
+				else
+				{	
+					placePiece(boardState, boardMove);
+				}
+
+			}
+		}
+		//4. Back-propagate results up to the root node once you get the results of your simulation.
+		//This just involves recording progress in the numWins and numSimulations members of nodes and 
+		//following parent up the tree till you hit NULL.
+		
+		 while(newNode->parent !=NULL)
+		 {
+		 	sum = 0;
+		 	numWins = 0;
+			numSims = 0;	
+		 	parentNode = newNode->parent;
+		 	//make sure that the parentNode has children
+		 	if(parentNode->numChildren > 0)
+		 	{
+		 		for(i = 0; i<parentNode->numChildren; i++)
+		 		{
+		 	
+		 			temp = parentNode->children[i];
+		 			numWins += temp->numWins;
+		 			numSims += temp->numSimulations;
+		 		}
+		 		sum = numWins + parentNode->numWins;
+		 		parentNode->numWins = sum;
+		 		sum = numSims + parentNode->numSimulations;
+		 		parentNode->numSimulations = sum;
+		 	}
+		 	newNode = parentNode;
+
+		 }
+
+		// check how much time has passed
+		timeElapsed = clock() - startTime;
+		if(timeElapsed > timeLimit)
+		{
+			break;
+		}
+		//check if there are any possible moves to make
+		if(root->numChildren == 0)
+		{
+			flag =-100;
+		}
+  	}
+
+  	bestValue = -20;
+  	numSims = -20;
+    index = -1;
+    //check that there were moves to be made
+    if(root->numChildren != 0)
+    {
+  		//loops to find the node that was most visited
+  		for(i = 0; i < root->numChildren; i++)
+		{
+			child = root->children[i];
+			numSims = child->numSimulations;
+			//finds the most visited node to choose the move to make
+		 	if(numSims > bestValue)
```

Comment:
So calculate UCT of the child node first, then figure out if it's the one with the highest UCT value.